client/src/telemetry/glean-context.tsx

```diff
@@ -93,3 +96,22 @@ export function GleanProvider(props: { children: React.ReactNode }) {
 export function useGlean() {
   return React.useContext(GleanContext);
 }
+
+export function useGleanPage() {
+  const loc = useLocation();
+  const isServer = useIsServer();
+
+  let referrer, location;
+  if (!isServer) {
+    referrer = document?.referrer;
+    location = window?.location;
+  }
+  return useEffect(
+    () =>
+      gleanAnalytics.page({
+        path: window?.location.toString(),
+        referrer: document?.referrer,
+      }),
+    [referrer, location, loc.pathname]
+  );
+}
```

Comment:
I'm a little confused by `referrer` and `location` being defined, and then not used in the glean call, and I think the way you're filtering out server-side events is causing the hook to be called multiple times if components are loaded and unloaded (as they are in local dev to prepare for react doing some fancy caching stuff in v19 or something).

How about something like this?

```js
export function useGleanPage() {
  const isServer = useIsServer();
  const routerLocation = useLocation();

  const referrer = document?.referrer;
  const location = window?.location.toString();

  return useEffect(() => {
    if (!isServer) {
      gleanAnalytics.page({
        path: location,
        referrer,
      });
    }
  }, [isServer, routerLocation.pathname, referrer, location]);
}
```

Reply:
Good call out. I think initially I had some issues with document/window being undefined and this was a work around. these globals are excluded from being required in the dependency array anyways.